client/rest/src/test/java/org/elasticsearch/client/SyncResponseListenerTests.java

```diff
@@ -124,27 +137,71 @@ public void testOnFailure() throws Exception {
         }
     }
 
-    public void testRuntimeExceptionIsNotWrapped() throws Exception {
+    public void testRuntimeIsBuiltCorrectly() throws Exception {
         RestClient.SyncResponseListener syncResponseListener = new RestClient.SyncResponseListener(10000);
         RuntimeException runtimeException = new RuntimeException();
         syncResponseListener.onFailure(runtimeException);
         try {
             syncResponseListener.get();
             fail("get should have failed");
         } catch(RuntimeException e) {
-            assertSame(runtimeException, e);
+            // We preserve the original exception in the cause
+            assertSame(runtimeException, e.getCause());
+            // We copy the message
+            assertEquals(runtimeException.getMessage(), e.getMessage());
+            // And we do all that so the thrown exception has our method in the stacktrace
+            assertExceptionStackContainsCallingMethod(e);
+        }
+    }
+
+    public void testConnectTimeoutExceptionIsBuiltCorrectly() throws Exception {
+        RestClient.SyncResponseListener syncResponseListener = new RestClient.SyncResponseListener(10000);
+        ConnectTimeoutException timeoutException = new ConnectTimeoutException();
+        syncResponseListener.onFailure(timeoutException);
+        try {
+            syncResponseListener.get();
+            fail("get should have failed");
+        } catch(IOException e) {
```

Comment:
nit : missing space 
s/catch(IOException e)/catch (IOException e)/
I see this at  several places in this file. Maybe reformat the new code ?

Reply:
I'll reform on the files that I touched I think.